pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -368,14 +368,14 @@ func NewController(kubeClient kubelib.Client, options Options) *Controller {
 
 	// This is for getting the node IPs of a selected set of nodes
 	c.nodeInformer = kubeClient.KubeInformer().Core().V1().Nodes().Informer()
-	_ = c.nodeInformer.SetTransform(stripNodeUnusedFields)
 	c.nodeLister = kubeClient.KubeInformer().Core().V1().Nodes().Lister()
 	nodeInformer := informer.NewFilteredSharedIndexInformer(nil, c.nodeInformer)
+	_ = c.nodeInformer.SetTransform(stripNodeUnusedFields)
```

Comment:
not sure i understand

Reply:
The podInformer/nodeInformer are passed into the `NewFilteredSharedIndexInformer` where we will reset transform as common `kube.StripUnusedFields`. We need to do the custom transform for pod/node after the `NewFilteredSharedIndexInformer`. Otherwise, the previous set operation will be overridden by the common `kube.StripUnusedFields`.

Reply:
I noticed that